src/hardware/serialport/nullmodem.cpp

```diff
@@ -406,44 +418,48 @@ void CNullModem::handleUpperEvent(Bit16u type) {
 		case SERIAL_NULLMODEM_DTR_EVENT: {
 			if ((!DTR_delta) && getDTR()) {
 				// DTR went positive. Try to connect.
-				if (ClientConnect(new TCPClientSocket((char*)hostnamebuffer,
-								(Bit16u)clientport)))
-					break; // no more DTR wait event when connected
-			}
-			DTR_delta = getDTR();
+			        if (ClientConnect(new TCPClientSocket((char *)hostnamebuffer,
+			                                              clientport)))
+				        break; // no more DTR wait event when
+				               // connected
+		        }
+		        DTR_delta = getDTR();
 			setEvent(SERIAL_NULLMODEM_DTR_EVENT,50);
 			break;
 		}
-	}
+	        }
 }
 
 /*****************************************************************************/
 /* updatePortConfig is called when emulated app changes the serial port     **/
 /* parameters baudrate, stopbits, number of databits, parity.               **/
 /*****************************************************************************/
-void CNullModem::updatePortConfig (Bit16u /*divider*/, Bit8u /*lcr*/) {
-	
-}
+void CNullModem::updatePortConfig(uint16_t /*divider*/, uint8_t /*lcr*/)
+{}
 
 void CNullModem::updateMSR () {
 	
 }
 
 bool CNullModem::doReceive () {
-		Bits rxchar = readChar();
-		if (rxchar>=0) {
-			receiveByteEx((Bit8u)rxchar,0);
-			return true;
-		}
-		else if (rxchar==-2) {
-			Disconnect();
-		}
-		return false;
+	uint8_t val;
+	SocketState state = readChar(val);
+	if (state == SocketState::Good) {
+		receiveByteEx(val, 0);
+		return true;
+	}
+	if (state == SocketState::Closed) {
+		Disconnect();
+	}
+	// socket was either empty or closed
+	return false;
```

Comment:
This function returns a `bool`, and internally we now get back a `SocketState`, so we assess this and maintain the same true/false return paths.